crawler/hosting.go

```diff
@@ -14,32 +14,66 @@ type Hosting struct {
 		ReqH int `yaml:"req/h"`
 		ReqM int `yaml:"req/m"`
 	} `yaml:"rate-limit"`
+	BasicAuth string `yaml:"basic-auth"`
 
 	ServiceInstance Crawler
 }
 
 // Repository is a single code repository.
 type Repository struct {
-	Name string
-	URL  string
+	Name    string
+	URL     string
+	Source  string
+	Headers map[string]string
 }
 
 // ParseHostingFile parses the hosting file to build a slice of Hosting.
 func ParseHostingFile(data []byte) ([]Hosting, error) {
 	hostings := []Hosting{}
 
-	// Unmarshal the yml in hostings list
-	err := yaml.Unmarshal(data, &hostings)
+	// Redis connection.
+	redisClient, err := redisClientFactory("redis:6379")
 	if err != nil {
 		return nil, err
 	}
 
+	// Unmarshal the yml in hostings list.
+	err = yaml.Unmarshal(data, &hostings)
+	if err != nil {
+		return nil, err
+	}
+
+	// Manage every host
 	for i, hosting := range hostings {
 		switch hosting.ServiceName {
 		case "bitbucket":
-			hostings[i].ServiceInstance = Bitbucket{
+			// Check if there is an URL that wasn't correctly retrieved.
+			// URL.value="false" => set hosting.URL to the one that one ("false")
+			keys, _ := redisClient.Keys("*").Result()
```

Comment:
how can I understand which hosting has been interrupted? Maybe we can use an hashmap to store the hosting name